app/controllers/posts_controller.rb

```diff
@@ -45,22 +47,19 @@ def owed
 
   def unread
     @started = (params[:started] == 'true') || (params[:started].nil? && current_user.unread_opened)
-    @posts = Post.joins("LEFT JOIN post_views ON post_views.post_id = posts.id AND post_views.user_id = #{current_user.id}")
-    @posts = @posts.joins("LEFT JOIN board_views on board_views.board_id = posts.board_id AND board_views.user_id = #{current_user.id}")
+    @posts = Post.not_ignored_by(current_user)
     @posts = @posts.where.not(post_views: { read_at: nil }) if @started
 
-    # post view does not exist and (board view does not exist or post has updated since non-ignored board view read_at)
-    no_post_view = @posts.where(post_views: { user_id: nil })
+    # post view does not exist and (board view does not exist or post has updated since board view read_at)
+    no_post_view = @posts.where(post_views: { id: nil })
     updated_since_board_read = no_post_view.where(board_views: { read_at: nil })
       .or(no_post_view.where("date_trunc('second', board_views.read_at) < date_trunc('second', posts.tagged_at)"))
-      .where(board_views: { ignored: false })
     no_post_view = no_post_view.where(board_views: { user_id: nil }).or(updated_since_board_read)
 
-    # post view exists and post has updated since non-ignored post view read_at and (board view does not exist or is not ignored)
-    with_post_view = @posts.where(post_views: { ignored: false }) # non-existant post-views will return nil here
-    with_post_view = with_post_view.where(board_views: { user_id: nil }).or(with_post_view.where(board_views: { ignored: false }))
-    with_post_view = with_post_view.where(post_views: { read_at: nil })
-      .or(with_post_view.where("date_trunc('second', post_views.read_at) < date_trunc('second', posts.tagged_at)"))
+    # post view exists and post has updated since post view read_at
+    with_post_view = @posts.where.not(post_views: { id: nil })
+    with_post_view = with_post_view.where(post_views: { read_at: nil }) # possible if someone ignores and then unignores a post, I think
+    with_post_view = with_post_view.or(@posts.where("date_trunc('second', post_views.read_at) < date_trunc('second', posts.tagged_at)"))
```

Comment:
```suggestion
    with_post_view = with_post_view.where(post_views: { read_at: nil }) # possible if someone ignores and then unignores a post
      .or(with_post_view.where("date_trunc('second', post_views.read_at) < date_trunc('second', posts.tagged_at)"))
```
This should be equivalent (the `@posts.where(date_trunc...)` will give NULL if post_views doesn't exist, which is falsy) but makes more sense in my brain, and is less of a change from how the code was before - of posts that have a view, we want either the ones where there's no read_at date, or the ones where they were read before the latest tag.

With the condition changed like this, `@posts` is all not-ignored-posts, not just the ones with views, so the logic of the condition is more complex to be sure of